packages/create-sitecore-jss/src/templates/nextjs-personalize/src/components/CdpIntegrationScript.tsx

```diff
@@ -1,89 +1,66 @@
-import { useSitecoreContext } from '@sitecore-jss/sitecore-jss-nextjs';
-import Script from 'next/script';
+import { useSitecoreContext, CdpHelper } from '@sitecore-jss/sitecore-jss-nextjs';
 import { useEffect } from 'react';
 import config from 'temp/config';
+import { init } from '@sitecore/engage';
 
-declare const _boxeverq: { (): void }[];
-declare const Boxever: Boxever;
-
-interface Boxever {
-  getID(): string;
-  eventCreate(data: BoxeverViewEventArgs, callback: () => void, format: string): void;
-}
-
-interface BoxeverViewEventArgs {
-  browser_id: string;
-  channel: string;
-  type: string;
-  language: string;
-  page: string;
-  pos: string;
-}
-
-function createPageView(locale: string, routeName: string) {
-  // POS must be valid in order to save events (domain name might be taken but it must be defined in CDP settings)
-  const pointOfSale =
-    process.env.NEXT_PUBLIC_CDP_POINTOFSALE || window.location.host.replace(/^www\./, '');
-
-  _boxeverq.push(function () {
-    const pageViewEvent: BoxeverViewEventArgs = {
-      browser_id: Boxever.getID(),
-      channel: 'WEB',
-      type: 'VIEW',
-      language: locale,
-      page: routeName,
-      pos: pointOfSale,
-    };
-
-    Boxever.eventCreate(
-      pageViewEvent,
-      function () {
-        /*empty callback*/
-      },
-      'json'
-    );
-  });
-}
-
+/**
+ * This is the CDP integration script which uses the Sitecore Engage SDK.
+ * It is used to enable page view events.
+ * See Sitecore Engage SDK documentation for details.
+ * https://www.npmjs.com/package/@sitecore/engage
+ */
 const CdpIntegrationScript = (): JSX.Element => {
   const {
-    sitecoreContext: { pageEditing, route },
+    sitecoreContext: { pageEditing, route, variantId },
   } = useSitecoreContext();
 
+  /**
+   * Creates a page view event using the Sitecore Engage SDK.
+   */
+  const createPageView = async (page: string, language: string, pageVariantId: string) => {
+    const engage = await init({
+      clientKey: process.env.NEXT_PUBLIC_CDP_CLIENT_KEY || '',
+      targetDomain: process.env.NEXT_PUBLIC_CDP_API_DOMAIN || '',
+      // Replace with the top level cookie domain of the website that is being integrated e.g ".example.com" and not "www.example.com"
+      cookieDomain: window.location.host.replace(/^www\./, ''),
+      // Cookie may be created in personalize middleware (server), but if not we should create it here
+      forceServerCookieMode: false,
+    });
```

Comment:
This may be temporary from your side but it's probably going to be more efficient to have initialization run less often

Reply:
Yeah, if a customer (or us, in the future) were using engage in more places it would make sense to place this somewhere else (e.g. a React Context) so that it can be re-used. However, for now it's meant to be as condensed/concise as possible for an example. Also, we can't easily introduce changes to higher levels (e.g. `[[...path]].tsx`, `_app.tsx`, or `Layout.tsx`) as we have to consider how that affects the base `nextjs` template and other add-ons (e.g. `nextjs-sxa`).

Reply:
In that case it's fine and you can probably point them towards the Engage samples for nextjs which will hopefully show a bigger range of example use cases.